### PR TITLE
feat: Add bandwidth stats to `PeerManagerActor` pt2

### PR DESCRIPTION
**Pull request by pmnoxx**
Continuation of https://github.com/near/nearcore/pull/5455

We are in progress of determining, what are the safe ranges of limit thresholds to set for `ThrottleController`.

`num_messages_in_progress` is part of `ThrottleController`. It's associated with given `PeerActor`. I tracks the number of messages which are currently in progress of being processed.

I added stats, which help us track, what was the maximum value of `num_messages_in_progress` across all peers over period of 1m. This will help us setup the liimits.

### CODE REVIEW

File: utils/near-rate-limiter/src/framed_read.rs [hunk 1]
```diff
@@ -46,8 +44,14 @@ pin_project! {
 pub struct ThrottleController {
     /// Total count of all messages that are tracked by `RateLimiterHelper`
     num_messages_in_progress: Arc<AtomicUsize>,
+    /// Maximum value of `num_messages_in_progress` recorded since last `consume` call.
+    max_record_num_messages_in_progress: Arc<AtomicUsize>,
     /// Total size of all messages that are tracked by `RateLimiterHelper`
     total_sizeof_messages_in_progress: Arc<AtomicUsize>,
+    /// Stores amount of bandwidth used.
+    bandwidth_used: Arc<AtomicUsize>,
```
Comment (mm-near) [inline utils/near-rate-limiter/src/framed_read.rs:52]: bandwidth -- is bytes sent and received?

File: utils/near-rate-limiter/src/framed_read.rs [hunk 2]
```diff
@@ -368,4 +399,26 @@ mod tests {
...
+        throttle_controller.report_msg_seen();
+        assert_eq!(throttle_controller.consume_msg_seen(), 3);
+        assert_eq!(throttle_controller.consume_msg_seen(), 0);
+
+        assert_eq!(throttle_controller.consume_bandwidth_used(), 0);
+        throttle_controller.report_bandwidth_used(10);
+        throttle_controller.report_bandwidth_used(1000);
+        throttle_controller.report_bandwidth_used(5);
+        assert_eq!(throttle_controller.consume_bandwidth_used(), 1015);
+        assert_eq!(throttle_controller.consume_bandwidth_used(), 0);
```
Comment (mm-near) [inline utils/near-rate-limiter/src/framed_read.rs:422]: can you also test max messages in progress?

File: utils/near-rate-limiter/src/framed_read.rs [hunk 3]
```diff
@@ -368,4 +399,26 @@ mod tests {
 
         assert_eq!(semaphore.available_permits(), 1);
     }
+
+    #[test]
```
Comment (mm-near) [inline utils/near-rate-limiter/src/framed_read.rs:403]: do we have any test for FrameReader? if not - can you add a very basic one, and see that throttler results are what we expect?

**Review by mm-near — APPROVED** [review]

File: utils/near-rate-limiter/src/framed_read.rs (hunk 1 above)
Comment (pmnoxx) [reply utils/near-rate-limiter/src/framed_read.rs]: received.

File: utils/near-rate-limiter/src/framed_read.rs (hunk 3 above)
Comment (pmnoxx) [reply utils/near-rate-limiter/src/framed_read.rs]: @mm-near I created a new PR for adding test: https://github.com/near/nearcore/pull/5758
It requires more work, I need to figure out why the test fails.

**Pull request merged**